test/e2e/proxy_test.go

```diff
@@ -169,18 +167,12 @@ func (tc *testContext) testEnvVarRemoval(t *testing.T) {
 	)
 	patchString := string(patchData)
 	require.NoErrorf(t, err, "unable to patch %s", patchString)
-	err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
-	assert.NoError(t, err, "timed out waiting for Windows Machine nodes")
-	err = tc.waitForConfiguredWindowsNodes(gc.numberOfBYOHNodes, false, true)
-	assert.NoError(t, err, "timed out waiting for BYOH Windows nodes")
 	for _, node := range gc.allNodes() {
 		t.Run(node.GetName(), func(t *testing.T) {
-			err := wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (done bool, err error) {
-				foundNode, err := tc.client.K8s.CoreV1().Nodes().Get(context.TODO(), node.GetName(), meta.GetOptions{})
-				require.NoError(t, err)
-				return tc.nodeReadyAndSchedulable(*foundNode), nil
-			})
-			assert.NoError(t, err)
+			err = tc.waitForRebootAnnotation(node)
+			require.NoError(t, err)
+			err = tc.waitForRebootAnnotationRemoval(node)
+			require.NoError(t, err)
```

Comment:
I was looking to the do the same sort of thing (for the cert removal test) and may see a timing issue with this approach. let me know what you think

let's say there are 2 nodes, A and B. we iterate over each node -- let's say our loop processes node A first. 
But what if WMCO reconciles node B first? While we're waiting in our loop for node A, node B could be fully reconciled (anno applied, node rebooted, anno removed). Then when our loop processes node B, it will timeout as the reboot anno will never be re-applied

Reply:
I stubbed out a function to solve this
```
// waitForRebootAnnotation waits for the reboot annotation to be applied on the node
func (tc *testContext) waitForAllNodesToReboot(nodeName string) error {
	// list all windows nodes, call it allWinNodes
	// have empty node list, call it seenWinNodes
	// keep listing all windows nodes with the reboot annotation -- until seenWinNodes has all the Nodes as allWinNodes
	// when any are seen, add them to seenWinNodes (if not already in the set)

	//then repeat above to wait for reboot anno to be removed from windows nodes
}
```

Reply:
Good catch, @saifshaikh48. I think the initial list has to be populated before the annotations are applied to guarantee that we don't miss anything. So in the context of the e2e test it will have to happen before the proxy configuration is changed.

Reply:
BTW I [implemented this func](https://github.com/openshift/windows-machine-config-operator/pull/1834/commits/a8a11d668eb952b154c9fb1639aa6673634b9f66#diff-5b235b7114a754117b4d8e52a9e9f2eb639dfe5a6930a130f2c5fb7826490295R118) as part of my testing

Reply:
updated please review.